warp10/src/main/java/io/warp10/script/functions/FOREACH.java

```diff
@@ -65,8 +74,11 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     }
     
     if (obj instanceof List) {
-      for (Object o: ((List<Object>) obj)) {
-        stack.push(o);
+      for (int i = 0; i < ((List) obj).size(); i++) {
```

Comment:
Why not keep the for (Object o: ...) loop and use an additional variable to use as the counter.